api/applicationserver_web.proto

```diff
@@ -101,6 +102,19 @@ message ApplicationWebhookTemplates {
   repeated ApplicationWebhookTemplate templates = 1;
 }
 
+message ApplicationWebhookHealth {
+  message WebhookHealthStatusHealthy { }
+  message WebhookHealthStatusUnhealthy {
+    uint64 failed_attempts = 1;
+    google.protobuf.Timestamp last_failed_attempt_at = 2 [(validate.rules).timestamp.required = true];
+    google.protobuf.Any last_failed_attempt_details = 3 [(validate.rules).any.required = true];
```

Comment:
How is a frontend going to display this? Why wouldn't this be error details or a string (but not any of them)?

Reply:
The data of events is marshaled in the same way, using `Any`, so we already have logic available that walks through the `Any` object and checks if it is indeed an `ErrorDetails` (and even knows how to walk through the `cause` field recursively), or at least an error string.

Reply:
I would prefer `ErrorDetails` indeed.

> so we already have logic available that walks through [...]

When you say "we" do you mean TTS? I'm thinking from the perspective of third party API clients; what do they do with `Any`?

Reply:
Third party clients could have used the `@type` field in order to determine how to handle the error, and that is consistent with most event handling. But indeed, let's make this one easy to use.

https://github.com/TheThingsNetwork/lorawan-stack/pull/4913/commits/30cc798654c0531369cd6e1ab4d1e0b380e1cd28